src/commands/variables/create.ts

```diff
@@ -29,29 +35,73 @@ export default class CreateVariable extends CreateCommand {
 
     public async run(): Promise<void> {
         const { flags } = await this.parse(CreateVariable)
-        const { key, name, type, feature, headless, project } = flags
+        const { key, name, type, feature, headless, project, variations } = flags
         await this.requireProject(project, headless)
 
-        if (headless && (!key || !name || !type || !feature)) {
-            this.writer.showError('The key, name, feature, and type flags are required')
+        if (headless && (!key || !name || !type)) {
+            this.writer.showError('The key, name, and type flags are required')
             return
         }
         flags._feature = feature
+        if (headless && variations && !flags._feature) {
+            this.writer.showError('Cannot modify variations without associating to a feature')
+            return
+        }
 
-        let params = await this.populateParametersWithZod(CreateVariableDto, this.prompts, flags)
+        const params = await this.populateParametersWithZod(CreateVariableDto, this.prompts, flags)
         if (!headless) {
             const { associateToFeature } = await inquirer.prompt([{
                 name: 'associateToFeature',
                 message: 'Would you like to associate this variable to a feature?',
                 type: 'confirm',
                 default: false
             }])
-    
+
             if (associateToFeature) {
-                params = await this.populateParametersWithZod(CreateVariableDto, [variableFeaturePrompt], params)
+                const { feature } = await inquirer.prompt([featurePrompt], {
+                    token: this.authToken,
+                    projectKey: this.projectKey
+                })
+                if (!feature) {
+                    this.writer.showError(`Feature with key ${feature.key} could not be found`)
+                    return
+                }
+                feature.variables.push(params)
+                const variableListOptions = new VariableListOptions(
+                    [params as Variable],
+                    this.writer,
+                    !variations ? feature.variations: undefined // don't prompt for variations if flag provided
+                )
+                await variableListOptions.promptVariationValues(params as Variable)
+                await updateFeature(this.authToken, this.projectKey, feature.key, feature)
+                const message = `The variable was associated to the existing feature ${feature.key}. ` +
+                    `Use "dvc feature get ${feature.key}" to see its details`
```

Comment:
should be `dvc features get --keys=${feature.key}`, i don't think we support `get <key>` currently (although we should)

Reply:
also should we still output the variable result here as well as the success message? might be kind of awkward since we'd have to find the new variable in the result of `updateFeature` but i feel like it might be annoying to have to get the feature separately just to check if you made a typo or something

Reply:
tbh I dont really think outputting the full variable is that useful here. Adam suggested that this be the response and I think I agree with him. Checking for a type is pretty much the only use case (especially since the command doesnt support headless mode if youre creating a feature)

Reply:
I will change the command thats output though, good catch!

Reply:
makes sense, works for me!